discovery-infra/tests/base_test.py

```diff
@@ -40,17 +40,22 @@ def nodes(self, setup_node_controller, request):
         else:
             node_vars = env_variables
         net_asset = None
+        needs_nat = node_vars.get("platform") == consts.Platforms.NONE
         try:
             if not qe_env:
                 net_asset = NetworkAssets()
                 node_vars["net_asset"] = net_asset.get()
             controller = setup_node_controller(**node_vars)
             nodes = Nodes(controller, node_vars["private_ssh_key_path"])
             nodes.prepare_nodes()
+            if needs_nat:
+                nodes.configure_nat()
```

Comment:
Why not put in its own fixture?

Reply:
Because it races with nodes

Reply:
So isn't it just the order of the fixtures which is the order of parameters?

Reply:
Both, If NAT was a standalone fixture it must have been called after nodes fixture, in this case I used it inside nodes fixture right after initializing the Nodes instance. The other option is to use it in the test itself with try finally block.
I see this implementation the best approach of the three

Reply:
OK